test/fakeintake/server/server.go

```diff
@@ -103,6 +107,16 @@ func WithClock(clock clock.Clock) func(*Server) {
 	}
 }
 
+func WithRetention(retention time.Duration) func(*Server) {
+	return func(fi *Server) {
+		if fi.URL() != "" {
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
Not related to this PR, we could add a `fi.IsRunning` that wraps `fi.URL` to make it more explicit

Reply:
I can do this here